### ISSUE TITLE
LDAP Source: No random active server available in server pool

### ISSUE THREAD

**Issue by thephilippbusch** [goauthentik/authentik#7559]
**Describe the bug**
When trying to connect a Microsoft Active Directory source with Authentik, the synchronization fails with an error code stating "no random active server available in server pool after maximum number of tries" (see logs below)

**To Reproduce**

1. Follow Active Directory setup from docs
2. Use settings as described below
3. Create new source
4. (optional: Use manual sync command for debugging)
5. See error

**Expected behavior**
Successful synchronization with active directory source

**Screenshots**
<img width="1149" alt="image" src="https://github.com/goauthentik/authentik/assets/47823894/d94e37b7-c966-411d-a30f-b167ada46e30">

<img width="1149" alt="image" src="https://github.com/goauthentik/authentik/assets/47823894/4febd0dc-f8ef-45db-a766-d263c7846794">

<img width="1149" alt="image" src="https://github.com/goauthentik/authentik/assets/47823894/c98c3baf-94c4-4d6d-8c10-5a15ef134505">

**Logs**

**Kubectl Logs:**
```
{"event": "Loaded config", "level": "debug", "logger": "authentik.lib.config", "timestamp": 1699943327.535007, "file": "/authentik/lib/default.yml"}
{"event": "Loaded environment variables", "level": "debug", "logger": "authentik.lib.config", "timestamp": 1699943327.5359437, "count": 48}
{"event": "Loaded app settings", "level": "debug", "logger": "authentik.lib.config", "timestamp": 1699943330.1464536, "path": "authentik.admin.settings"}
{"event": "Loaded app settings", "level": "debug", "logger": "authentik.lib.config", "timestamp": 1699943330.1475837, "path": "authentik.crypto.settings"}
{"event": "Loaded app settings", "level": "debug", "logger": "authentik.lib.config", "timestamp": 1699943330.1484487, "path": "authentik.events.settings"}
{"event": "Loaded app settings", "level": "debug", "logger": "authentik.lib.config", "timestamp": 1699943330.1498187, "path": "authentik.outposts.settings"}
{"event": "Loaded app settings", "level": "debug", "logger": "authentik.lib.config", "timestamp": 1699943330.152013, "path": "authentik.policies.reputation.settings"}
{"event": "Loaded app settings", "level": "debug", "logger": "authentik.lib.config", "timestamp": 1699943330.1561882, "path": "authentik.providers.scim.settings"}
{"event": "Loaded app settings", "level": "debug", "logger": "authentik.lib.config", "timestamp": 1699943330.1576564, "path": "authentik.sources.ldap.settings"}
{"event": "Loaded app settings", "level": "debug", "logger": "authentik.lib.config", "timestamp": 1699943330.1584861, "path": "authentik.sources.oauth.settings"}
{"event": "Loaded app settings", "level": "debug", "logger": "authentik.lib.config", "timestamp": 1699943330.1591902, "path": "authentik.sources.plex.settings"}
{"event": "Loaded app settings", "level": "debug", "logger": "authentik.lib.config", "timestamp": 1699943330.1644654, "path": "authentik.stages.authenticator_totp.settings"}
{"event": "Loaded app settings", "level": "debug", "logger": "authentik.lib.config", "timestamp": 1699943330.1688643, "path": "authentik.blueprints.settings"}
{"event": "Booting authentik", "level": "info", "logger": "authentik.lib.config", "timestamp": 1699943330.1690793, "version": "2023.10.3"}
{"event": "Enabled authentik enterprise", "level": "info", "logger": "authentik.lib.config", "timestamp": 1699943330.1703951}
{"event": "Loaded app settings", "level": "debug", "logger": "authentik.lib.config", "timestamp": 1699943330.171004, "path": "authentik.enterprise.settings"}
{"event": "Loaded GeoIP database", "last_write": 1699552276.0, "level": "info", "logger": "authentik.events.geo", "pid": 150, "timestamp": "2023-11-14T06:28:51.644033"}
{"event": "Task published", "level": "info", "logger": "authentik.root.celery", "pid": 150, "task_id": "17e44aca4c1c471b8b7594a6c3eff71e", "task_name": "authentik.blueprints.v1.tasks.blueprints_discovery", "timestamp": "2023-11-14T06:28:52.995831"}
{"event": "Task published", "level": "info", "logger": "authentik.root.celery", "pid": 150, "task_id": "ac7302c6916440368d375d88a9c56a63", "task_name": "authentik.blueprints.v1.tasks.clear_failed_blueprints", "timestamp": "2023-11-14T06:28:53.002724"}
{"app_name": "authentik.stages.authenticator", "event": "Could not import app's URLs", "exc": "ModuleNotFoundError(\"No module named 'authentik.stages.authenticator.urls'\")", "level": "warning", "logger": "authentik.api.v3.urls", "pid": 150, "timestamp": "2023-11-14T06:28:53.254078"}
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/manage.py", line 31, in <module>
    execute_from_command_line(sys.argv)
  File "/ak-root/venv/lib/python3.11/site-packages/django/core/management/__init__.py", line 442, in execute_from_command_line
    utility.execute()
  File "/ak-root/venv/lib/python3.11/site-packages/django/core/management/__init__.py", line 436, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/ak-root/venv/lib/python3.11/site-packages/django/core/management/base.py", line 412, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/ak-root/venv/lib/python3.11/site-packages/django/core/management/base.py", line 458, in execute
    output = self.handle(*args, **options)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/authentik/sources/ldap/management/commands/ldap_sync.py", line 27, in handle
    ldap_sync_paginator(source, UserLDAPSynchronizer)
  File "/authentik/sources/ldap/tasks.py", line 76, in ldap_sync_paginator
    sync_inst: BaseLDAPSynchronizer = sync(source)
                                      ^^^^^^^^^^^^
  File "/authentik/sources/ldap/sync/base.py", line 30, in __init__
    self._connection = source.connection()
                       ^^^^^^^^^^^^^^^^^^^
  File "/authentik/sources/ldap/models.py", line 176, in connection
    successful = connection.bind()
                 ^^^^^^^^^^^^^^^^^
  File "/ak-root/venv/lib/python3.11/site-packages/ldap3/core/connection.py", line 589, in bind
    self.open(read_server_info=False)
  File "/ak-root/venv/lib/python3.11/site-packages/ldap3/strategy/sync.py", line 57, in open
    BaseStrategy.open(self, reset_usage, read_server_info)
  File "/ak-root/venv/lib/python3.11/site-packages/ldap3/strategy/base.py", line 122, in open
    new_server = self.connection.server_pool.get_server(self.connection)  # get a server from the server_pool if available
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/ak-root/venv/lib/python3.11/site-packages/ldap3/core/pooling.py", line 317, in get_server
    return self.pool_states[connection].get_server()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/ak-root/venv/lib/python3.11/site-packages/ldap3/core/pooling.py", line 97, in get_server
    self.last_used_server = self.find_active_random_server()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/ak-root/venv/lib/python3.11/site-packages/ldap3/core/pooling.py", line 143, in find_active_random_server
    raise LDAPServerPoolExhaustedError('no random active server available in server pool after maximum number of tries')
ldap3.core.exceptions.LDAPServerPoolExhaustedError: no random active server available in server pool after maximum number of tries
command terminated with exit code 1
```

**Version and Deployment (please complete the following information):**

-   authentik version: 2023.10.3
-   Deployment: helm

**Additional context**

Helm values.yaml:
```
authentik:
  secret_key: "some secret..."
  error_reporting:
    enabled: false
  postgresql:
    password: "a password"
    host: "postgres-service"
    user: "authentik"
    name: "authentik"
  redis:
    password: "a password"
    host: "redis-master"

ingress:
  ingressClassName: traefik 
  enabled: true
  hosts:
    - host: authentik.test.example.com
      paths:
        - path: "/"
          pathType: Prefix

postgresql:
  enabled: true
  postgresqlPassword: "a password"
redis:
  enabled: true
  auth:
    enabled: true
```

Connection to active directory/ldap server via specified URLs and configuration should work fine and was tested with ldapsearch. I did not yet test the functionality with other/older versions.

**Comment by thephilippbusch** [comment]
**TL;DR**
Setting the server URI-protocols to "ldaps://" while leaving all other options as is did the trick in my case.

I was able to fix the bug by trying further configuration variants. Setting the server URIs to "ldaps://" protocol leaving all other options as is did the trick in my case. Maybe a more specific error message could help the debugging process, as I was quite unsure what to do with the message I got. But I guess there not much one can do when it comes to LDAP connection problems of that sort.
Anyway, maybe this is useful for other people that run into similar issues.
